### DISCUSSION TITLE
installation fail v8 in php 7 laravel 7

### DISCUSSION THREAD

**Discussion by SurajTiwari007** [spatie/laravel-medialibrary#1849] [Q&A]
Error
  Class name must be a valid object or a string

  at C:\xampp\htdocs\UjitaJobPortal\vendor\spatie\laravel-medialibrary\src\MediaLibraryServiceProvider.php:23
    19|         $this->registerPublishables();
    20|
    21|         $mediaClass = config('media-library.media_model');
    22|
  > 23|         $mediaClass::observe(new MediaObserver());
    24|
    25|         $this->loadViewsFrom(__DIR__.'/../resources/views', 'media-library');
    26|     }
    27|

  1   C:\xampp\htdocs\UjitaJobPortal\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:32
      Spatie\MediaLibrary\MediaLibraryServiceProvider::boot()

  2   C:\xampp\htdocs\UjitaJobPortal\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:32
      call_user_func_array([])
Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1

Installation failed, reverting ./composer.json to its original content.

**Comment by i-defranca** [comment]
same here

**Comment by Demers94** [comment]
I had the same problem and solved it by using `php artisan config:clear`

**Comment by SurajTiwari007** [comment]
> I had the same problem and solved it by using `php artisan config:clear`

i fired this command but error will be same

**Comment by alibarthi** [comment]
Hi,

Have you followed all the steps in the [Upgrade Guide](https://github.com/spatie/laravel-medialibrary/blob/master/UPGRADING.md)?

**Comment by hirenkeraliya** [comment]
@alibarthi  
The fresh installation also not works.

**Comment by chrisbjr** [comment]
Happened to me too after migrating from v7 to v8. I had to delete all the contents `bootstrap/cache` (except the .gitignore file). The error disappeared after this.

**Comment by jadonbrownza** [comment]
Can't require medialibrary in a new laravel 7 project. Getting the following before composer restores the original composer.json file.

```
    - Can only install one of: symfony/process[4.4.x-dev, v5.0.5].
    - Can only install one of: symfony/process[v4.4.0, v5.0.5].
    - Can only install one of: symfony/process[v4.4.0-BETA1, v5.0.5].
    - Can only install one of: symfony/process[v4.4.0-BETA2, v5.0.5].
    - Can only install one of: symfony/process[v4.4.0-RC1, v5.0.5].
    - Can only install one of: symfony/process[v4.4.1, v5.0.5].
    - Can only install one of: symfony/process[v4.4.2, v5.0.5].
    - Can only install one of: symfony/process[v4.4.3, v5.0.5].
    - Can only install one of: symfony/process[v4.4.4, v5.0.5].
    - Can only install one of: symfony/process[v4.4.5, v5.0.5].
    - Can only install one of: symfony/process[v4.4.6, v5.0.5].
    - Can only install one of: symfony/process[v4.4.7, v5.0.5].
    - Can only install one of: symfony/process[v4.4.8, v5.0.5].
    - Installation request for symfony/process (locked at v5.0.5) -> satisfiable by symfony/process[v5.0.5].
```

**Comment by JayAhrDe** [comment]
@scheMeZa Had the same error (that should not be related to the one of this issue, though ;) ) -- it worked after enabling gd2 and exif, that had not been enabled in _that_ installed PHP version... Found those by running `composer update` after failed `composer require`.

**Comment by MahbodHastam** [comment]
i saw all issues and forums... but still have this problem :/
